### ISSUE TITLE
A11Y: Keyboard navigation using tabs should be trapped when roles editor is open

### ISSUE THREAD

**Issue by eclarke1** [google/site-kit-wp#5440]
## Feature Description

Bug bash issue: https://app.asana.com/0/1202258919887896/1202443514146836 please see Asana for background and for screen recording.

When the user edits roles the focus should not leave the roles editor component. If the user hits the Tab key when the last pill is selected, then the focus should go to the close editor button instead of jumping to the cancel button in the footer.

I think having the close button next makes sense, but technically we shouldn't be focus trapping tab here… if anything the pills might make sense to be navigable with arrow inputs… but being able to eventually tab out of that area makes sense. But making close come up after the rest of the pills in the tab order is a good call. 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* In the Dashboard Sharing edit roles screen, the "Close" button should appear and be in a tab order _after_ all role chips.
* The visual position and order in the DOM of the close button and the Dashboard Sharing role select should be switched, essentially swapping these two sections:
![CleanShot 2022-06-28 at 15 38 11](https://user-images.githubusercontent.com/90871/176207273-858560f3-8edf-4ea9-9281-4391b4aa90db.png)
  * The close button should be in a consistent position regardless of the number of role chips; this probably means right-aligning it or using flex/grid to position it after the box that contains all role chips.
* When activating the Dashboard Sharing "edit roles" mode, the first selection should be the "All roles" chip, not the "Close" button as it currently is:
![CleanShot 2022-06-28 at 15 40 14](https://user-images.githubusercontent.com/90871/176207607-2f840e8b-cc3b-4b35-aa62-35f36504c9ed.gif)
* Moving the DOM order should be sufficient to also make the "Close" button come after in tab order. The reason we aren't only changing the tab order is because it is not good accessibility to place elements visually out-of-order with the tab order.

## Implementation Brief

* In `assets/js/components/dashboard-sharing/UserRoleSelect.js`:
    * Copy the `.googlesitekit-user-role-select__button` `<Button />` component and place it inside the check for `editMode` being truthy, after the `.googlesitekit-user-role-select__chipset` `<div>` (wrap the `<div>` and the new `<Button />` component with `<Fragment />`).
    https://github.com/google/site-kit-wp/blob/dcf1251865f75738260af419efb6f371235598d1/assets/js/components/dashboard-sharing/UserRoleSelect.js#L169-L186
    * Update the first original `<Button />` component to:
        * Show only if `editMode` is falsey.
        * The `aria-label` prop should only be `Edit roles` (translatable).
        * The `icon` prop should only contain the `<ShareIcon />` component.
    * Update the new `<Button />` component to:
        * The `aria-label` prop should only be `Close` (translatable).
        * The `icon` prop should only contain the `<CloseIcon />` component.

### Test Coverage

* Update VRT images if required.

## QA Brief

* Enable the `dashboardSharing` feature flag and open the Dashboard Settings modal.
* Edit the roles and check that the Close button visually appears after the roles.
* Check the keyboard navigation works as expected with the Close button coming after the roles for tab navigation.
* Verify the above in desktop and mobile viewports, and with/without multiple admins.

## Changelog entry

* Improve keyboard accessibility/behaviour when selecting user roles in Dashboard Sharing modal.

**Comment by tofumatt** [comment]
Looks good 👍🏻 

IB ✅

**Comment by mohitwp** [comment]
### QA Update ✅ 

Verified:

- Verified on Firefox. Chrome, Safari and edge latest versions.
- Verified for single as well as for multiple admin scenario.
- Verified on iPhone 13, iPad and Android mobile and tablet using BrowserStack.
- Keyboard navigation working as expected with the Close button coming after the roles for tab navigation.

<details>


https://user-images.githubusercontent.com/94359491/183400186-3b11ee25-d9ea-4231-aee3-536c52986e2a.mp4

![image](https://user-images.githubusercontent.com/94359491/183400230-2bfc9533-965e-4c28-a47f-398e4f21bbf3.png)

</details>
